spec/rubocop/cop/style/safe_navigation_spec.rb

```diff
@@ -1169,6 +1169,44 @@ def foobar
           RUBY
         end
 
+        it 'registers an offense for a safe navigation method call followed by a method call' do
+          expect_offense(<<~RUBY, variable: variable)
+            %{variable}&.bar && %{variable}.bar.baz
+            ^{variable}^^^^^^^^^^{variable}^^^^^^^^ Use safe navigation (`&.`) instead [...]
+          RUBY
+
+          expect_correction(<<~RUBY)
+            #{variable}&.bar&.baz
+          RUBY
+        end
+
+        it 'registers an offense for multiple method calls with safe navigation on last call followed by a method call' do
+          expect_offense(<<~RUBY, variable: variable)
+            %{variable}.bar&.baz && %{variable}.bar.baz.quux
+            ^{variable}^^^^^^^^^^^^^^{variable}^^^^^^^^^^^^^ Use safe navigation (`&.`) instead [...]
+          RUBY
+
+          expect_correction(<<~RUBY)
+            #{variable}.bar&.baz&.quux
+          RUBY
+        end
+
+        it 'registers offenses for an object check followed by a method call followed again by another method call' do
+          expect_offense(<<~RUBY, variable: variable)
+            %{variable} && %{variable}.bar && %{variable}.bar.baz
+            _{variable}    ^{variable}^^^^^^^^^{variable}^^^^^^^^ Use safe navigation (`&.`) instead [...]
+            ^{variable}^^^^^{variable}^^^^ Use safe navigation (`&.`) instead [...]
+          RUBY
+
+          expect_correction(<<~RUBY, loop: false)
+            #{variable}&.bar && #{variable}.bar.baz
+          RUBY
+
```

Comment:
Having only one `expect_correction` should be sufficient.

```suggestion
```

Reply:
My intention was to ensure that after the first pass the autocorrection is not `#{variable}.bar && #{variable}.bar.baz` (without safe nav), but I see now that we have other tests which cover this scenario, so I removed this expectation. Thanks.